### ISSUE TITLE
Allow passing parameters into types by name

### ISSUE THREAD

**Issue by dgelessus** [kaitai-io/kaitai_struct#690]
Currently parameters can only be passed positionally. This is fine for one or two parameters, but it can get confusing when there are many parameters, or if the data types are not very descriptive.

For example, when using the [`bcd`](https://github.com/kaitai-io/kaitai_struct_formats/blob/02be57c1b455138bdba99b684a131adbabb6977a/common/bcd.ksy) type from kaitai_struct_formats, you currently have to write something like `bcd(4, 4, false)`. As a reader it's not clear what this means - is the `4` a byte size? What is the other `4` for? The `false` at the end is also completely meaningless if you don't already know the `bcd` spec. It would be more readable if the parameters could be passed by name, like `bcd(num_digits=4, bits_per_digit=4, is_le=false)`.

This would also be useful in combination with #260 (default values for parameters). With named parameters you can selectively override any of the parameter defaults; with positional parameters you can only override defaults at the start of the argument list (to override a parameter at the end, you also have to pass all parameters that come before it).

**Comment by KOLANICH** [comment]
IMHO: if we ever pass named parameters, they should be passed as YAML.

**Comment by dgelessus** [comment]
@KOLANICH Do you mean something like this?

```
seq:
  - id: bcd_thing
    type: bcd
    params:
      num_digits: 4
      bits_per_digit: 4
      is_le: false
```

It helps a bit with readability, but I'm not a huge fan of the syntax difference compared to positional arguments. Also, this way it would be impossible to mix positional and named arguments (e. g. `bcd(4, is_le=true)`), which I think should be allowed.

**Comment by KOLANICH** [comment]
>Also, this way it would be impossible to mix positional and named arguments (e. g. bcd(4, is_le=true)), which I think should be allowed.

I don't think so. In fact it is possible the same way we define unnamed and named fields in `seq`, but I don't think it makes any sense. If we use named args, it means we have damn lot of args in the signature, so damn lot that we don't remember their order, and even if we do, we have troubles in counting commas.s

>It helps a bit with readability,

It's not about readability, it is about availability to third-party tools, like other impls of KSC and generators of ksy specs like synalisys2kaitai.
